portal/migrations/versions/e503ae1c261a_.py

```diff
@@ -0,0 +1,45 @@
+from alembic import op
+import sqlalchemy as sa
+
+
+"""empty message
+
+Revision ID: e503ae1c261a
+Revises: a031e26dc1bd
+Create Date: 2018-08-14 16:05:28.104848
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'e503ae1c261a'
+down_revision = 'a031e26dc1bd'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        'users',
+        sa.Column(
+            'last_password_verification_failure',
```

Comment:
In the future I would suggest using redis for storing data that could be considered ephemeral (flask-user similarly uses redis to store server-side session data). 

Redis also has the ability to [automatically expire keys](https://redis.io/commands/setex) which could slightly simplify the timeout logic

No need for any change- wouldn't be worth the effort to refactor

Reply:
Thanks for calling this out. When I started thinking about this work I didn't want to store it in a DB because it doesn't make perfect sense there, but didn't know where else to put it. Wish I had spoken with you last week! Will keep redis in mind for the future. I've used it in the past for other stuff. Would have been a great place to put it.